src/uct/ib/ud/base/ud_ep.c

```diff
@@ -372,6 +363,16 @@ static void uct_ud_ep_timer(ucs_wtimer_t *self)
         }
     }
 
+    diff = now - ep->tx.send_time;
+    if (diff > iface->config.peer_timeout) {
```

Comment:
why needed to move?

Reply:
is in required to make keepalive work or unrelated fix?

Reply:
got it, now that peer_timeout is a smaller value, we exposed the issue that it had to be checked last